packages/@webex/plugin-meetings/src/media/properties.js

```diff
@@ -195,4 +197,99 @@ export default class MediaProperties {
     this.unsetLocalVideoTrack();
     this.unsetRemoteMedia();
   }
+
+  /**
+   * Waits until ice connection is established
+   *
+   * @returns {Promise<void>}
+   */
+  waitForIceConnectedState() {
+    const isIceConnected = () => (
+      this.peerConnection.iceConnectionState === ICE_STATE.CONNECTED ||
+        this.peerConnection.iceConnectionState === ICE_STATE.COMPLETED
+    );
+
+    if (isIceConnected()) {
+      return Promise.resolve();
+    }
+
+    return new Promise((resolve, reject) => {
+      let timer;
+
+      const iceListener = () => {
+        LoggerProxy.logger.log(`Media:properties#waitForIceConnectedState --> ice state: ${this.peerConnection.iceConnectionState}, conn state: ${this.peerConnection.connectionState}`);
+
+        if (isIceConnected()) {
+          clearTimeout(timer);
+          this.peerConnection.removeEventListener('iceconnectionstatechange', iceListener);
+          resolve();
+        }
+      };
+
+      timer = setTimeout(() => {
+        this.peerConnection.removeEventListener('iceconnectionstatechange', iceListener);
+        reject();
+      }, PC_BAIL_TIMEOUT);
+
+      this.peerConnection.addEventListener('iceconnectionstatechange', iceListener);
+    });
+  }
+
+  /**
+   * Returns the type of a connection that has been established
+   *
+   * @returns {Promise<'UDP' | 'TCP' | 'TURN-TLS' | 'TURN-TCP' | 'TURN-UDP' | 'unknown'>}
+   */
+  async getCurrentConnectionType() {
+    // we can only get the connection type after ICE connection has been established
+    await this.waitForIceConnectedState();
+
+    const allStatsReports = [];
+
+    try {
+      // eslint-disable-next-line no-await-in-loop
+      const statsResult = await this.peerConnection.getStats();
+
+      statsResult.forEach((report) => allStatsReports.push(report));
```

Comment:
putting all statsReports into an array so that it's easier to work with (so I can filter them, etc)